File: backend/src/main/java/egovframework/let/production/plan/domain/model/ProductionPlan.java
package egovframework.let.production.plan.domain.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import org.apache.commons.lang3.builder.ToStringBuilder;

import java.io.Serializable;
import java.math.BigDecimal;

/**
 * 생산계획 상세 데이터 모델 클래스
 * @author SHMT-MES
 * @since 2025.11.21
 * @version 1.0
 * @see
 *
 * <pre>
 * << 개정이력(Modification Information) >>
 *
 *   수정일      수정자           수정내용
 *  -------    --------    ---------------------------
 *   2025.11.21 SHMT-MES          최초 생성
 *
 * </pre>
 */
@Schema(description = "생산계획 상세 모델")
@Getter
@Setter
public class ProductionPlan implements Serializable {

	/**
	 *  serialVersion UID
	 */
	private static final long serialVersionUID = 1L;

	@Schema(description = "회사 코드")
	private String factoryCode = "000001";

	@Schema(description = "생산계획ID")
	private String prodPlanId = "";

	@Schema(description = "생산계획일자 (YYYYMMDD)")
	private String prodPlanDate = "";

	@Schema(description = "생산계획순번")
	private int prodPlanSeq = 0;

	@Schema(description = "생산계획일자 (YYYYMMDD)")
	private String planDate = "";

	@Schema(description = "묶음 ID (생성일수 그룹)")
	private String planGroupId = "";

	@Schema(description = "묶음 내 순번")
	private Integer groupSeq = 0;

	@Schema(description = "생성일수")
	private Integer createDays = 1;

	@Schema(description = "품목코드")
	private String itemCode = "";

	@Schema(description = "품목명")
	private String itemName = "";

	@Schema(description = "계획수량")
	private BigDecimal plannedQty = BigDecimal.ZERO;

	@Schema(description = "실적수량")
	private BigDecimal actualQty = BigDecimal.ZERO;

	@Schema(description = "작업장 코드")
	private String workplaceCode = "";

	@Schema(description = "작업장명")
	private String workplaceName = "";

	@Schema(description = "공정 코드")
	private String processCode = "";

	@Schema(description = "공정명")
	private String processName = "";

	@Schema(description = "설비ID")
	private String equipmentId = "";

	@Schema(description = "설비코드")
	private String equipmentCode = "";

	@Schema(description = "설비명")
	private String equipmentName = "";

	@Schema(description = "근무구분 (A/B/C/D/N)")
	private String workerType = "";

	@Schema(description = "작업자 코드")
	private String workerCode = "";

	@Schema(description = "작업자명")
	private String workerName = "";

	@Schema(description = "LOT번호")
	private String lotNo = "";

	@Schema(description = "거래처 코드")
	private String customerCode = "";

	@Schema(description = "거래처명")
	private String customerName = "";

	@Schema(description = "납기일 (YYYYMMDD)")
	private String deliveryDate = "";

	// 주문(생산의뢰) 연동 정보 - TPR301R 저장용
	@Schema(description = "생산의뢰번호")
	private String orderNo = "";

	@Schema(description = "생산의뢰 순번")
	private Integer orderSeqno = 0;

	@Schema(description = "생산의뢰 이력번호")
	private Integer orderHistno = 0;

	@Schema(description = "비고")
	private String remark = "";

	@Schema(description = "사용 여부 (Y/N)")
	private String useYn = "Y";

	@Schema(description = "등록자 ID")
	private String opmanCode = "";

	@Schema(description = "등록일시")
	private String optime = "";

	@Schema(description = "수정자 ID")
	private String opmanCode2 = "";

	@Schema(description = "수정일시")
	private String optime2 = "";

	/**
	 * toString 메소드를 대치한다.
	 */
	public String toString(){
		return ToStringBuilder.reflectionToString(this);
	}
}

File: frontend/src/types/productionPlan.ts
// 생산계획 관련 공통 타입 (UI 전용 데이터 구조)
// 서비스 계층(ProductionPlan)과 달리 date(YYYY-MM-DD) 형식을 사용하며 UI 편의 필드(id 등)를 포함합니다.
export interface ProductionPlanData {
  id?: string; // UI에서 planNo-planSeq 조합 등으로 생성되는 식별자
  // 기본 계획 정보
  date: string; // YYYY-MM-DD (planDate 가공)
  itemCode: string;
  itemDisplayCode?: string; // UI 노출용 품목코드 (백엔드 저장용 itemCode와 분리)
  itemName: string;
  plannedQty: number;
  equipmentId?: string;
  equipmentCode: string;
  equipmentName?: string;
  shift?: string; // DAY / NIGHT 등
  remark?: string;
  // 생성일수/묶음 정보
  createDays?: number;
  planGroupId?: string;
  groupSeq?: number;
  totalGroupCount?: number;
  // 생산의뢰 연동 정보
  orderNo?: string;
  orderSeqno?: number;
  orderHistno?: number;
  // 작업장/공정/작업자 정보
  workplaceCode?: string;
  workplaceName?: string;
  processCode?: string;
  processName?: string;
  workerCode?: string;
  workerName?: string;
  // 거래처 정보
  customerCode?: string;
  customerName?: string;
  additionalCustomers?: string[]; // 다수 거래처 표시용
  // 납기일
  deliveryDate?: string; // YYYY-MM-DD 또는 YYYYMMDD
  // 백엔드 원본 필드(선택적으로 활용)
  planNo?: string;
  planSeq?: number;
  factoryCode?: string;
  actualQty?: number;
  lotNo?: string;
  useYn?: string;
  // 상태 정보
  orderFlag?: string; // ORDERED: 생산지시 완료, PLANNED: 계획만 등록
}

export type ProductionPlanFormUpdates = Partial<ProductionPlanData>;
